spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducer.java

```diff
@@ -0,0 +1,207 @@
+	 */
+	public void setReadOffset(ReadOffset readOffset) {
+		this.readOffset = readOffset;
+	}
+
+	/**
+	 * Configure this channel adapter to extract or not the message payload.
+	 * @param extractPayload default true
+	 */
+	public void setExtractPayload(boolean extractPayload) {
+		this.extractPayload = extractPayload;
+	}
+
+	/**
+	 * Set the name of the Consumer Group. It is possible to create that Consumer Group if desired, see:
+	 * {@link #createConsumerGroupIfNotExist}. Note that if this value is set, {@link #consumerName} should also be set
+	 * @param consumerGroupName the Consumer Group which this adapter should register to listen messages.
+	 */
+	public void setConsumerGroupName(@Nullable String consumerGroupName) {
+		this.consumerGroupName = consumerGroupName;
+	}
+
+	/**
+	 * If the Consumer Group is used, this value should be provided. Note that this value should be unique in the group
+	 * @param consumerName the conusmer name in the Consumer Group
+	 */
+	public void setConsumerName(@Nullable String consumerName) {
+		this.consumerName = consumerName;
+	}
+
+	/**
+	 * Create the Consumer Group if and only if it does not exist. During the
+	 * creation we also create the stream {@see MKSTREAM}. If the stream already exists {@code MKSTREAM}
+	 * has no effect.
+	 * @param createConsumerGroupIfNotExist specify if we should create the Consumer Group, {@code false} by default
+	 */
+	public void setCreateConsumerGroupIfNotExist(boolean createConsumerGroupIfNotExist) {
+		this.createConsumerGroupIfNotExist = createConsumerGroupIfNotExist;
+	}
+
+	/**
+	 * Set {@link StreamOperations} used to customize the {@link StreamReceiver}.
+	 * It provides a way to set the polling timeout and the serialization context.
+	 * By default the polling timeout is set to infinite and {@link StringRedisSerializer} is used.
+	 * @param streamReceiverOptions the desired receiver options
+	 * */
+	public void setStreamReceiverOptions(@Nullable StreamReceiver.StreamReceiverOptions<String, ?> streamReceiverOptions) {
+		this.streamReceiverOptions = streamReceiverOptions;
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		if (this.streamReceiverOptions != null) {
+			this.streamReceiver = StreamReceiver.create(this.reactiveConnectionFactory, this.streamReceiverOptions);
+		}
+		else {
+			this.streamReceiver = StreamReceiver.create(this.reactiveConnectionFactory);
+		}
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+	}
+
+	@Override
+	protected void doStart() {
+		super.doStart();
+
+		this.streamKey = this.streamKeyExpression.getValue(this.evaluationContext, String.class);
+
+		Assert.notNull(this.streamKey, "'streamKey' must not be null");
+
+		StreamOffset<String> offset = StreamOffset.create(this.streamKey, this.readOffset);
+
+		Flux<Message<?>> messageFlux;
+
+		if (StringUtils.isEmpty(this.consumerGroupName)) {
+			messageFlux = this.streamReceiver
+					.receive(offset)
```

Comment:
See this API:
```
* Every record must be acknowledged using
* {@link org.springframework.data.redis.connection.ReactiveStreamCommands#xAck(ByteBuffer, String, String...)}
```
Therefore we need to have a ` boolean autoAck` option on this channel adapter and consider to build an `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK` as a `SimpleAcknowledgment` implementation.

Reply:
@artembilan I think I will need your help for this. Also, I though that acknowledgement was only for the case where Consumer Group was involved.

Reply:
Yeah... Looks like you are right. Only consumer group needs an acknowledge: https://redis.io/topics/streams-intro.
Also see `ReactiveStreamOperations.acknowledge()` API: all of them require a `group` to ack the record.

Probably Spring Data team made a copy/paste mistake to mention ack on the regular `receive()`:
```
/**
 * Starts a Redis Stream consumer that consumes {@link Record records} from the {@link StreamOffset stream}. Records
 * are consumed from Redis and delivered on the returned {@link Flux} when requests are made on the Flux. The receiver
 * is closed when the returned {@link Flux} terminates.
 * <p/>
 * Every record must be acknowledged using
 * {@link org.springframework.data.redis.connection.ReactiveStreamCommands#xAck(ByteBuffer, String, String...)}
 *
 * @param streamOffset the stream along its offset.
 * @return Flux of inbound {@link Record}s.
 * @see StreamOffset#create(Object, ReadOffset)
 */
Flux<V> receive(StreamOffset<K> streamOffset);
```

Reply:
The `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK` could be as simple as:
```
.setHeader(IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK, 
                     () -> this.redisStreamOperations.acknowledge(this.consumerGroup, event))
```